api/controllers/NetworkController.js

```diff
@@ -1,7 +1,62 @@
 module.exports = {
 
-  findOne: function(req, res) {
-    res.ok(res.locals.network);
-  }
+  findOne: function (req, res) {
+    res.ok(res.locals.network)
+  },
+
+  create: function (req, res) {
+    var attrs = _.pick(req.allParams(),
+      'name', 'description', 'slug', 'banner_url', 'avatar_url')
+
+    var network = new Network(_.merge(attrs, {
+      created_at: new Date()
+    }))
+
+    return network.save()
+    .tap(network => Promise.map(req.param('communities'), communityId =>
+      Membership.hasModeratorRole(req.session.userId, communityId)
+      .then(isModerator => {
+        if (isModerator) {
+          return Community.find(communityId)
+          .then(community => community.save({network_id: network.id}))
+        }
+        return
+      })
+    ))
+    .then(res.ok)
+    .catch(res.serverError)
+  },
+
+  validate: function (req, res) {
+    var allowedColumns = ['name', 'slug']
```

Comment:
this code looks familiar :) want to pull it out of here and CommunityController into a new file in services? there's already a good set of tests for it